src/components/line-chart-audification/line-chart-audification.component.ts

```diff
@@ -1,25 +1,96 @@
-import { Component, Host } from '@angular/core';
+import { Component, Host, Input, OnDestroy, OnInit } from '@angular/core';
 import { LineChartComponent } from '../line-chart/line-chart.component';
-import { AudificationService } from '../../services/audification/audification.service';
-import * as Tone from 'tone';
+import { Subscription } from 'rxjs';
+import { Melody } from '../../models/melody/melody.model';
 
 @Component({
   selector: 'app-line-chart-audification',
   templateUrl: './line-chart-audification.component.html',
 })
-export class LineChartAudificationComponent {
-  public disposeMelody?: () => void;
+export class LineChartAudificationComponent implements OnInit, OnDestroy {
+  @Input() frequencyRange: [number, number] = [256, 2048];
+  @Input() duration = 5;
+  private element = this.component.element.nativeElement;
+  private lineChartD3 = this.component.lineChartD3;
+  private melody?: Melody;
+  private dataSubscription?: Subscription;
+  private refocusTimeout?: number;
 
   constructor(
     @Host() public component: LineChartComponent,
-    private audificationService: AudificationService,
   ) {
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.handleKeyUp = this.handleKeyUp.bind(this);
   }
 
-  async playMelody() {
-    if (Tone.getContext().state === 'suspended') {
-      await Tone.start();
+  ngOnInit() {
+    this.dataSubscription = this.component.dataObservable.subscribe(data => {
+      const values = data.map(datum => datum.value);
+      this.melody?.dispose();
+      this.melody = new Melody(values, this.frequencyRange, this.duration, (index, playing) => {
+        this.component.activeDatum = data[index];
+        if (!playing) {
+          this.readOut(this.lineChartD3.activeGroupId);
+        }
+      });
+    });
+    this.element.addEventListener('keydown', this.handleKeyDown);
+    this.element.addEventListener('keyup', this.handleKeyUp);
+  }
+
+  ngOnDestroy() {
+    this.element.removeEventListener('keyup', this.handleKeyUp);
+    this.element.removeEventListener('keydown', this.handleKeyDown);
+    this.dataSubscription?.unsubscribe();
+    this.melody?.dispose();
+    if (this.refocusTimeout !== undefined) {
+      window.clearTimeout(this.refocusTimeout);
+    }
+  }
+
+  async handleKeyDown($event: KeyboardEvent) {
+    const { key, shiftKey, repeat } = $event;
+    if (repeat) {
+      return;
+    }
+    if (key === ' ') {
```

Comment:
I don't think I'll be adding more shortcuts for the same action in order to minimize conflicts with screen-reader applications. (Speaking of which, it might be a good idea to let users customize shortcuts.) I'll go with `if` for now, but I'll consider using `switch ... case` when I implement the shortcut customization.